dvc/commands/plots.py

```diff
@@ -72,27 +71,35 @@ def run(self):
                 )
 
             renderers = match_renderers(
-                plots_data=plots_data, templates=self.repo.plots.templates
+                plots_data=plots_data, out=self.args.out
             )
 
             if self.args.show_vega:
                 renderer = first(
-                    filter(lambda r: isinstance(r, VegaRenderer), renderers)
+                    filter(lambda r: r.TYPE == "vega", renderers)
                 )
                 if renderer:
-                    ui.write_json(renderer.asdict())
+                    ui.write_json(json.loads(renderer.partial_html()))
                 return 0
             if self.args.json:
```

Comment:
IMO, `--json` should be returning the raw `plots_data` . 

This would be consistent with `exp show --json` and would prevent us from maintaining and updating conversion logic.

Reply:
I agree, thought on the other hand I would presume that users expect the same plots for DVC and vs code, even more than they would expect same plots for DVC and studio. And we already had some problems [there](https://github.com/iterative/viewer/issues/2574). @mattseddon what would you prefer more? Having datapoints created on our side and just filling the template with them or make to data -> datapoints conversion yourself?

Reply:
Let's leave this behavior as is for now. `--json` is not official, if VSCode will prefer to do the data processing by themselves, we will change it.